src/modifier.ts

```diff
@@ -167,3 +173,178 @@ function pairwiseMerge(modifiers1: Modifiers, modifiers2: Modifiers) {
+          }
+          printModtrace(modifiersSubset, keyThis);
+        }
+      });
+    } else {
+      let htmlOutput = cliExecuteOutput(`modtrace ${baseModifier}`);
+      let htmlHeader = htmlOutput.substring(
+        htmlOutput.indexOf("<tr>") + 4,
+        htmlOutput.indexOf("</tr>")
+      );
+      let headers = [] as string[];
+      let headerMatches = htmlHeader.match("(>)(.*?)(</td>)");
+      while (headerMatches) {
+        const header = headerMatches[2];
+        headers.push(header);
+
+        const idx =
+          headerMatches[0].length + htmlHeader.search("(>)(.*?)(</td>)");
+        htmlHeader = htmlHeader.substring(idx);
+        headerMatches = htmlHeader.match("(>)(.*?)(</td>)");
+      }
+      headers = headers.slice(2);
+
+      const exactModifierColIdx = headers.findIndex(
+        (header) => header.toLowerCase() === baseModifier.toLowerCase()
+      );
+
+      if (exactModifierColIdx === -1) {
+        print(
+          `Could not find exact string match of ${baseModifier} in ${modifiers.toString()}`,
+          "red"
+        );
+        return;
+      }
+
+      let totalVal = 0.0;
+      // Maps modifier name to its value
+      const modifierVals = new Map(
+        headers.map((header) => {
+          return [header, 0];
+        })
+      );
+
+      const lowerCaseModifiers = modifiers.map((modifier) =>
+        modifier.toLowerCase()
+      );
+
+      if (baseModifier.toLowerCase() === "familiar weight") {
+        totalVal += familiarWeight(myFamiliar());
+        print(`[Familiar Weight] Base weight (${totalVal})`);
+      }
+
+      htmlOutput = htmlOutput.substring(
+        htmlOutput.indexOf("</tr>") + 5,
+        htmlOutput.indexOf("</table>")
+      );
+
+      while (htmlOutput.length > 0) {
+        const idxStart = htmlOutput.indexOf("<tr>");
+        const idxEnd = htmlOutput.indexOf("</tr>");
+        if (idxStart === -1) break;
+
+        let row = replaceAll(
+          htmlOutput.substring(idxStart + 4, idxEnd),
+          "></td>",
+          ">0</td>"
+        );
+        const rowArr = [] as string[];
+        let rowMatches = row.match("(>)(.*?)(</td>)");
+        while (rowMatches) {
+          rowArr.push(rowMatches[2]);
+          row = row.replace(rowMatches[0], "");
+          rowMatches = row.match("(>)(.*?)(</td>)");
+        }
+        rowArr
+          .slice(2)
+          .filter((e, idx) => idx % 2 === 0)
+          .forEach((e, idx) => {
+            const val = parseInt(e);
+            modifierVals.set(
+              headers[idx],
+              (modifierVals.get(headers[idx]) ?? 0) + val
+            );
+            if (
+              val !== 0 &&
+              lowerCaseModifiers.includes(headers[idx].toLowerCase())
+            ) {
+              print(`[${headers[idx]}] ${rowArr[1]} (${val.toFixed(1)})`);
+            }
+          });
+
+        htmlOutput = htmlOutput.substring(idxEnd + 5);
+      }
+
+      let total = 0.0;
```

Comment:
I would move to an Array.reduce paradigm instead of using these `let`s with `for` loops